src/api.rs

```diff
@@ -176,10 +178,25 @@ pub enum Registers {
 pub const PAGE_SHIFT: u32 = 12;
 pub const PAGE_SIZE: u32 = 4096;
 
+#[derive(thiserror::Error, Debug)]
+pub enum MicrovmiError {
+    #[error(transparent)]
+    DriverError(#[from] DriverError),
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum DriverError {
```

Comment:
This kinda bums me out. We have to wrap errors from each driver that implements the `Introspectable` trait which essentially breaks abstraction. :/